apps/src/codebridge/InfoPanel/index.tsx

```diff
@@ -0,0 +1,99 @@
+import React, {useState} from 'react';
+
+import Button from '@cdo/apps/componentLibrary/button';
+import InstructionsView from '@cdo/apps/lab2/views/components/Instructions';
+import PanelContainer from '@cdo/apps/lab2/views/components/PanelContainer';
+import {useAppSelector} from '@cdo/apps/util/reduxHooks';
+
+import HelpAndTips from './HelpAndTips';
+
+import moduleStyles from './styles/info-panel.module.scss';
+
+enum Panels {
+  Instructions = 'Instructions',
+  HelpAndTips = 'Help and Tips',
+}
+
+export const InfoPanel = React.memo(() => {
+  const mapReference = useAppSelector(
+    state => state.lab.levelProperties?.mapReference
+  );
+  const referenceLinks = useAppSelector(
+    state => state.lab.levelProperties?.referenceLinks
+  );
+  const [currentPanel, setCurrentPanel] = useState(Panels.Instructions);
+  const [isDropdownOpen, setIsDropdownOpen] = useState(false);
+
+  const renderHeaderButton = () => {
+    return panelOptions().length > 1 ? (
+      <div>
+        <Button
+          icon={{
+            iconStyle: 'solid',
+            iconName: isDropdownOpen ? 'caret-up' : 'caret-down',
+          }}
+          isIconOnly
+          onClick={() => setIsDropdownOpen(!isDropdownOpen)}
+          color={'black'}
+          ariaLabel={'Information panel dropdown'}
+          size={'xs'}
+        />
+      </div>
+    ) : null;
+  };
+
+  const panelOptions = () => {
+    // For now, always include Instructions panel.
+    // TODO: support hiding this panel completely if there are no instructions.
+    const panelOptions = [Panels.Instructions];
+    if (mapReference || referenceLinks) {
+      panelOptions.push(Panels.HelpAndTips);
+    }
+    return panelOptions;
+  };
```

Comment:
Does this need to be a function or could `panelOptions` just be defined as an array? Is there a need to recalculate the options each time `panelOptions()` is called or would it be sufficient to do it once?

Going a step further, would it be worth memoizing `panelOptions` so that it only re-calculates when there are changes to `mapReference` or `referenceLinks`?

Reply:
The reason to keep it as a function is we could switch levels and go from having to not having help and tips, and want to update accordingly.
It's a pretty cheap operation, so I don't think we need to memoize. The [react docs](https://react.dev/reference/react/useMemo#skipping-expensive-recalculations) say you don't need to memoize for cheap operations.

Reply:
Sounds good to me! These hooks are still pretty new to me, so I've been trying to test my understanding of how (and why) to use them. Thanks for explaining!

Reply:
I ended up changing `panelOptions` to be a state value instead, because I found a bug where if you switch levels and were previously on a panel that does not exist on the new level it will still show as your selected panel, but be blank. So I needed to add a `useEffect` that depended on the panel options changing.